src/components/inputs/text/NameTokenSelector/NameTokenSelector.tsx

```diff
@@ -0,0 +1,591 @@
+import { Pagination, PaginationProps, Tooltip } from 'antd';
+import { useEffect, useRef, useState } from 'react';
+
+import { useArweaveCompositeProvider } from '../../../../hooks';
+import { useGlobalState } from '../../../../state/contexts/GlobalState';
+import {
+  ArweaveTransactionID,
+  PDNTContractJSON,
+  VALIDATION_INPUT_TYPES,
+} from '../../../../types';
+import { isArweaveTransactionID } from '../../../../utils';
+import eventEmitter from '../../../../utils/events';
+import {
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  CircleCheck,
+  CirclePlus,
+  CircleXIcon,
+} from '../../../icons';
+import { Loader } from '../../../layout';
+import ValidationInput from '../ValidationInput/ValidationInput';
+
+function NameTokenSelector({
+  walletAddress,
+  selectedTokenCallback,
+}: {
+  walletAddress: ArweaveTransactionID;
+  selectedTokenCallback: (id: ArweaveTransactionID | undefined) => void;
+}) {
+  const arweaveDataProvider = useArweaveCompositeProvider();
+  const [{ pdnsSourceContract }] = useGlobalState();
+
+  const [searchText, setSearchText] = useState<string>();
+  const [tokens, setTokens] = useState<{
+    [x: string]: Partial<PDNTContractJSON>;
```

Comment:
`Partial` is too flexible, if you only want to include certain values use `Pick<PDNTContarctJSON>` or `Omit<PDNTContractjSON>`

Reply:
and `nameTokens` might be more clear here - ChatGPT tokens comes to mind when i see this 😆